### PR TITLE
[zos/jobs] Added method to hold/release a job

### LINKED ISSUE #181: [zos-jobs] Add methods to hold/release a job
Add a method to the Python zos-jobs SDK to call the z/OSMF REST API `PUT /zosmf/restjobs/jobs/<jobname>/<jobid>` with `hold` or `release` request

* z/OSMF REST API documentation:
    * https://www.ibm.com/docs/en/zos/2.5.0?topic=interface-hold-job
    * https://www.ibm.com/docs/en/zos/2.5.0?topic=interface-release-job
* Python zos-jobs SDK in the "main" branch: https://github.com/zowe/zowe-client-python-sdk/tree/main/src/zos_jobs
* Node.js zos-jobs SDK for comparison: https://github.com/zowe/zowe-cli/blob/master/packages/zosjobs/src/ModifyJobs.ts

### PR DESCRIPTION
**Pull request by aadityasinha-dotcom**
**What It Does**
Added methods to `hold/release` a job
Resolves #181 

**How to Test**

**Review Checklist**
I certify that I have:
- [x] tested my changes
- [x] added/updated automated tests
- [ ] updated the changelog
- [x] followed the [contribution guidelines](https://github.com/zowe/zowe-cli/blob/master/CONTRIBUTING.md)

**Additional Comments**

### CODE REVIEW

**Review by t1m0thyj — CHANGES_REQUESTED** [review]
Thanks @aadityasinha-dotcom 🙂 Left a few comments about reducing code duplication and merging integration tests

File: src/zos_jobs/zowe/zos_jobs_for_zowe_sdk/jobs.py [hunk 1]
```diff
@@ -116,6 +116,72 @@ def delete_job(self, jobname, jobid, modify_version="2.0"):
...
+        request_url = "{}{}".format(self.request_endpoint, job_url)
+        custom_args["url"] = request_url
+        custom_args["json"] = {
+            "request": "release",
+            "version": modify_version
+        }
+        custom_args["headers"]["X-IBM-Job-Modify-Version"] = modify_version
+
+        response_json = self.request_handler.perform_request("PUT", custom_args, expected_code=[202, 200])
+        return response_json
```
[the remark this answers is not in the document]
Comment (aadityasinha-dotcom) [reply src/zos_jobs/zowe/zos_jobs_for_zowe_sdk/jobs.py:184]: Then where do we call this utility function? In the seperate method for hold and release job?

[the remark this answers is not in the document]
Comment (t1m0thyj) [reply src/zos_jobs/zowe/zos_jobs_for_zowe_sdk/jobs.py]: Yes, implementation for hold and release jobs methods looks good 👍 

Please update `change_jobs_class` to also call the utility function 😋

**Comment by aadityasinha-dotcom** [comment]
@t1m0thyj please review the changes :relaxed:

File: src/zos_jobs/zowe/zos_jobs_for_zowe_sdk/jobs.py [hunk 2]
```diff
@@ -128,10 +128,20 @@ def _issue_job_request(self, req: dict, jobname: str, jobid: str, modify_version
...
-            custom_args["headers"]["X-IBM-Job-Modify-Version"] = modify_version
+        elif "class" in req:
+            custom_args = self._create_custom_request_arguments()
+            job_url = "{}/{}".format(jobname, jobid)
+            request_url = "{}{}".format(self.request_endpoint, job_url)
+            custom_args["url"] = request_url
+            custom_args["json"] = {
+                "class": req["class"],
+                "version": modify_version 
+            }
```
Comment (t1m0thyj) [inline src/zos_jobs/zowe/zos_jobs_for_zowe_sdk/jobs.py:139]: I think lines 122-139 could be simplified more still to remove the if-else branches 🙂 
```
    custom_args = self._create_custom_request_arguments()
    job_url = "{}/{}".format(jobname, jobid)
    request_url = "{}{}".format(self.request_endpoint, job_url)
    custom_args["url"] = request_url
    custom_args["json"] = {
        **req,
        "version": modify_version 
    }
```

File: src/zos_jobs/zowe/zos_jobs_for_zowe_sdk/jobs.py [hunk 3]
```diff
@@ -117,6 +117,22 @@ def delete_job(self, jobname, jobid, modify_version="2.0"):
...
+            **req,
+            "version": modify_version 
+        }
+
+        custom_args["headers"]["X-IBM-Job-Modify-Version"] = modify_version
+            
+        response_json = self.request_handler.perform_request("PUT", custom_args, expected_code=[202, 200])
+        return response_json
+       
     def change_jobs_class(self, jobname: str, jobid: str, class_name: str, modify_version="2.0"):
```
Comment (t1m0thyj) [inline src/zos_jobs/zowe/zos_jobs_for_zowe_sdk/jobs.py:136]: Could you please rename the method to `change_job_class` ("job" singular)? Sorry for being picky 😢

Comment (aadityasinha-dotcom) [reply src/zos_jobs/zowe/zos_jobs_for_zowe_sdk/jobs.py]: Done with changes :relaxed:

**Review by t1m0thyj — APPROVED** [review]

**Review by zFernand0 — APPROVED** [review]

**Pull request merged**